src/main/java/net/minecraftforge/common/ForgeModContainer.java

```diff
@@ -381,32 +386,88 @@ private static void syncConfig(boolean load)
         }
     }
 
+    /**
+     * @return the order of hud config entries, represented by their keys
+     */
+    private static ArrayList<String> setupHudConfig(String categoryName)
+    {
+        //config names, with their "friendly name" used for comment generating
+        final String[][] allConfigs = new String[][]{
+            {"allowVignette", "vignette"},
+            {"allowHelmet", "helmet"},
+            {"allowPortal", "portal"},
+            {"allowHotbar", "hotbar"},
+            {"allowCrosshairs", "crosshairs"},
+            {"allowBossHealth", "boss health"},
+            {"allowHealth", "health"},
+            {"allowArmor", "armor"},
+            {"allowFood", "food"},
+            {"allowHealthMount", "health mount"},
+            {"allowAir", "air"},
+            {"allowExperience", "experience"},
+            {"allowJumpBar", "jump bar"},
+            {"allowObjective", "objective"},
+        };
+
+        var order = new ArrayList<String>(allConfigs.length);
+        var values = new boolean[allConfigs.length];
+
+        for (int i = 0; i < values.length; i++)
+        {
+            var key = allConfigs[i][0];
+            var friendlyName = allConfigs[i][1];
+            var comment = "Toggle off to force disable %s rendering. Note that this config is applied on config saved, mods can still change %s rendering behaviour after that."
```

Comment:
`Toggle off to disable rendering of the %s. Note that even though this config is applied when the config is saved to disk, mods can still override the behaviour after-the-fact.`